### PR TITLE
Refactor/combine

### PR DESCRIPTION
**Pull request by Brooooooklyn**
### Description
重新定义 `combine` 并将其重命名为 `concat`，老的 `combine` 实现保留。

### Related
https://github.com/teambition/ReactiveDB/issues/5

### CODE REVIEW

File: src/storage/RuntimeError.ts [hunk 1]
```diff
@@ -97,8 +97,11 @@ export const INVALID_PATCH_TYPE_ERR =
 export const TOKEN_CONSUMED_ERR =
   () => new ReactiveDBError('QueryToken was already consumed.')
 
-export const TOKEN_INVALID_ERR =
-  () => new ReactiveDBError(`Token cannot be combined.`)
+export const TOKEN_CONCAT_ERR =
+  (msg?: string) => {
+    const errMsg = `Token cannot be concat ${ msg ? `due to: ${ msg }` : '' }`
```
Comment (Saviio) [inline src/storage/RuntimeError.ts]:
```
'Token cannot be concated' + `${ msg ? ' due to: ' + msg : '' }.`
```

File: src/storage/Selector.ts [hunk 2]
```diff
@@ -172,11 +191,19 @@ export class Selector <T> {
...
 
   combine(... selectMetas: Selector<T>[]): Selector<T> {
-    const isEqual = selectMetas.every(meta => meta.select === this.select)
-    if (!isEqual) {
-      throw TOKEN_INVALID_ERR()
+    return Selector.combineFactory(this, ... selectMetas)
+  }
+
+  concat(... selectMetas: Selector<T>[]): Selector<T> {
+    const equal = selectMetas.every(m =>
```
Comment (Saviio) [inline src/storage/Selector.ts]: consider
```
SELECT * FROM TABLE WHERE id >=0 SKIP 1 LIMIT 10
SELECT * FROM TABLE WHERE id >=0 SKIP 2 LIMIT 20
...
```

these query will violate constraint of `concat` method, maybe we should add a term to check if `skip mod limit` is equal to `0`

File: src/storage/Selector.ts [hunk 3]
```diff
@@ -26,8 +26,27 @@ export interface OrderInfo {
 }
 
 export class Selector <T> {
-  static factory<U>(... metaDatas: Selector<U>[]) {
-    const originalToken = metaDatas[0]
+  static concatFactory<U>(... metaDatas: Selector<U>[]) {
+    const [ meta ] = metaDatas
+    const skips = metaDatas
+      .map(m => m.skip)
+      .sort()
```
Comment (Saviio) [inline src/storage/Selector.ts]: consider:
```sql
SELECT * FROM TABLE WHER id >= 0 SKIP 10 LIMIT 10
SELECT * FROM TABLE WHER id >= 0 SKIP 20 LIMIT 10
SELECT * FROM TABLE WHER id >= 0 SKIP 30 LIMIT 10
...
SELECT * FROM TABLE WHER id >= 0 SKIP 100 LIMIT 10
```
after sort => [10, 100, 20, 30]

Comment (chuan6) [inline src/storage/Selector.ts]: 这边 `skip` 是数字不是字符串吧？

`private skip?: number`

File: src/storage/Selector.ts (hunk 2 above)
Comment (Brooooooklyn) [inline src/storage/Selector.ts]: limit should be equal here

**Pull request merged**